src/wrench/services/compute/hadoop/DeterministicMRJob.cpp

```diff
@@ -8,13 +8,66 @@
  */
 
 #include "wrench/services/compute/hadoop/DeterministicMRJob.h"
+#include "wrench/services/compute/hadoop/hadoop_subsystem/Mapper.h"
 #include "wrench/logging/TerminalOutput.h"
 
 WRENCH_LOG_NEW_DEFAULT_CATEGORY(hadoop_compute_servivce, "Log category for Deterministic MR Job");
 
 namespace wrench {
 
-    DeterministicMRJob::DeterministicMRJob() : MRJob() {
+    DeterministicMRJob::DeterministicMRJob(int num_mappers, int num_reducers, double data_size,
+                                           bool use_combiner, int sort_factor, double spill_percent,
+                                           int key_value_width) : MRJob() {
+        this->setNumMappers(num_mappers);
+        this->setNumReducers(num_reducers);
+        this->setUseCombiner(use_combiner);
+        this->setDataSize(data_size);
+        this->setSortFactor(sort_factor);
+        this->setSpillPercent(spill_percent);
+        this->setKeyValueWidth(key_value_width);
+        this->setJobType(std::string("deterministic"));
+    }
+
+    // Typical set of parameters
+    DeterministicMRJob::DeterministicMRJob() {
+        this->setNumMappers(10);
+        this->setNumReducers(1);
+        this->setUseCombiner(false);
+        this->setDataSize(1048576);  // 2^20 bytes
+        this->setSortFactor(10);
+        this->setSpillPercent(0.8);
+        this->setKeyValueWidth(32);
+        this->setJobType(std::string("deterministic"));
+    };
+
+    DeterministicMRJob::~DeterministicMRJob() = default;
+
+    std::pair<double, double> DeterministicMRJob::getMapCost(double map_function_cost) {
+        /** TODO:
+         * Using the deterministic model, the map cost is as simple
+         * as:
+         *    num_mappers * total_map_cost
+         * A Mapper object will have the following
+         * sub-costs:
+         *  - Read: Via an `HDFS` object
+         *  - User map function cost: Unsure
+         *  - Collection and spill: Via a `Spill` object
+         *  - Merge: Via a `MapSideMerge` object
+         */
+
+        auto mapper = new Mapper(*this, map_function_cost);
```

Comment:
Curious if this is the right way for an object's function to self reference?

Reply:
Sure. i mean, passing "this" to the constructor of an object so that it has a ref to its creator is fine.